modules/tactical/src/tactical_store.ts

```diff
@@ -0,0 +1,92 @@
+/// <reference path="../../../typings/rx/rx.d.ts" />
+/// <reference path="../../../typings/rx/rx-lite.d.ts" />
+import {Observable} from 'rx';
+
+import {Idb} from './idb';
+import {Key} from './key';
+import {Version, VersionKeyPair, VersionEntityPair} from './version';
+
+
+export class TacticalStore {
+
+  private static _store: string = "tactical";
+  
+  constructor(private _idbRef: Idb) {}
+  
+  static get store(): string {
+    return this._store;
+  }
+  
+  fetch(key: Object, version?: string): Observable<VersionEntityPair> {
+    return Observable.create<VersionEntityPair>((observer: Rx.Observer<VersionEntityPair>) => {
+      if (version) {
+        this._idbRef.get(TacticalStore.store, (new VersionKeyPair(version, key)).serial).subscribe(
+          (entity: Object) => {
+            if (entity) {
+              observer.onNext(new VersionEntityPair(version, entity));
+            } else {
+              observer.onNext(null);
+            }
+          },
+          (err: any) => {
+            observer.onError(err);
+          }
+        );
+      } else {
+        // fetch the latest version of the object
+        this._idbRef.get(TacticalStore.store, (new Key(key)).serial).subscribe(
+          (ver: Version) => {
+            if (ver && ver.version) {
+              this._idbRef.get(TacticalStore.store, (new VersionKeyPair(ver.version, key)).serial)
+                  .subscribe(
+                (entity: Object) => {
+                  if (entity) {
+                    observer.onNext(new VersionEntityPair(ver.version, entity));
+                  } else {
+                    observer.onNext(null); 
+                  }
+                },
+                (err: any) => {
+                  observer.onError(err);
+                }
+              );
+            } else {
+              observer.onNext(null);
+            }
+          },
+          (err: any) => {
+            observer.onError(err);
+          }
+        );
+      }
+    });
+  }
+ 
+  commit(key: Object, entity: Object, version: string): Observable<boolean> {
```

Comment:
The key is serialized twice in one invocation of commit() (once as an effect of creating the VersionKeyPair, a second time as a result of creating the Key). See if you can avoid this.